src/actions/channel.js

```diff
@@ -98,13 +102,22 @@ export const loadChannelMembers = () => (dispatch, getState) => {
   })
 
   api
-    .listMembers(channel.id)
-    .then(res => res.results)
-    .then(users => users.map(normalizeUserData))
-    .then(payload => {
+    .listMembers(channel.id, {
+      limit: 50,
+      after,
+    })
+    .then(({ results, total }) => ({
+      users: results.map(normalizeUserData),
+      total,
+    }))
+    .then(({ users, total }) => {
+      const { users: loadedUsers } = channelMembersSelector(getState())
       dispatch({
         type: types.HANDLE_CHANNEL_MEMBERS,
-        payload,
+        payload: {
+          users: isInitialLoading ? users : [...loadedUsers, ...users],
+          total,
```

Comment:
@nikgraf  yes, if we already have users in the list, we will add more from `users`. Maybe `loadedUsers`, not a very good idea how to name members which already loaded in prev request. I'll think about it

Reply:
The name could be better, but wasn't my concern. Let me try to explain via an example:
```
currentState: {
  users: [{ id: 'a' }, { id: 'b'}]
  total: 2,
}
```

If you now get the response with

```
users: [{ id: 'a' } , { id: 'b' } , {id: 'c'}]
total: 3,
```

The final list will be:

```
users: [{ id: 'a' } , { id: 'b' }, { id: 'a' } , { id: 'b' }, {id: 'c'}]
total: 3,
```

Two issues here:
1. First you need to filter users by ID do avoid duplicates.
2. The total must be the correct number of actual users in that list?!?

Reply:
aaaahhh total is coming from the API as a value. Ignore my second point then. But still you need to filter out duplicates. Correct?

Reply:
@nikgraf I don't how it can happen, because request contains `after` param, so duplicates are not possible.

Reply:
@nikgraf total just needs inside component to know how to compare how many items already loaded and what's a total number of members.

Reply:
Just read up on the API and you are right in that case! Because it's ordered by joinedAt we might miss a member if someone leaves, but we won't get any duplicates.

In any case I still would recommend to make this a common practice when updating a reducer state. I a lot of cases it avoid unnecessary duplication bugs.